starknet_py/net/client.py

```diff
@@ -171,7 +170,7 @@ async def wait_for_tx(
                 elif status != TransactionStatus.RECEIVED:
                     # This will never get executed with current possible transactions statuses
                     raise TransactionFailedError(
-                        message=result.rejection_reason,
+                        message=cast(str, result.rejection_reason),
```

Comment:
Perhaps we should add a "getter" method for transaction rejection reason that always returns not None value, based on the transaction status? This way we can use it without this kind of type casting?